ibis/backends/pyspark/client.py

```diff
@@ -66,3 +300,458 @@ def execute(
+        node = self.table_class(qualified_name, schema, self)
+        return self.table_expr_class(node)
+
+    def list_tables(self, like=None, database=None):
+        """
+        List tables in the current (or indicated) database. Like the SHOW
+        TABLES command.
+
+        Parameters
+        ----------
+        like : string, default None
+          e.g. 'foo*' to match all tables starting with 'foo'
+        database : string, default None
+          If not passed, uses the current/default database
+
+        Returns
+        -------
+        results : list of strings
+        """
+        results = [t.name for t in self._catalog.listTables(dbName=database)]
+        if like:
+            results = [
+                table_name
+                for table_name in results
+                if re.match(like, table_name) is not None
+            ]
+
+        return results
+
+    def exists_table(self, name, database=None):
+        """
+        Determine if the indicated table or view exists
+
+        Parameters
+        ----------
+        name : string
+        database : string, default None
+
+        Returns
+        -------
+        if_exists : boolean
+        """
+        try:
+            self._get_jtable(name, database)
+            return True
+        except com.IbisInputError:
+            return False
+
+    def set_database(self, name):
+        """
+        Set the default database scope for client
+        """
+        self._catalog.setCurrentDatabase(name)
+
+    def list_databases(self, like=None):
+        """
+        List databases in the Spark SQL cluster.
+
+        Parameters
+        ----------
+        like : string, default None
+          e.g. 'foo*' to match all tables starting with 'foo'
+
+        Returns
+        -------
+        results : list of strings
+        """
+        results = [db.name for db in self._catalog.listDatabases()]
+        if like:
+            results = [
+                database_name
+                for database_name in results
+                if re.match(like, database_name) is not None
+            ]
+
+        return results
+
+    def exists_database(self, name):
+        """
+        Checks if a given database exists
+
+        Parameters
+        ----------
+        name : string
+          Database name
+
+        Returns
+        -------
+        if_exists : boolean
+        """
+        return bool(self.list_databases(like=name))
+
+    def create_database(self, name, path=None, force=False):
+        """
+        Create a new Spark database
+
+        Parameters
+        ----------
+        name : string
+          Database name
+        path : string, default None
+          Path where to store the database data; otherwise uses Spark
+          default
+        """
+        statement = CreateDatabase(name, path=path, can_exist=force)
+        return self.raw_sql(statement.compile())
+
+    def drop_database(self, name, force=False):
```

Comment:
Are these method moved from the original spark file?

Reply:
Yes, I moved everything that it was inherited from the base class, before removing it.